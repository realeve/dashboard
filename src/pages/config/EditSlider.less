@import '~@/style/theme.less';

.edit-slider {
  position: absolute;
  bottom: 0;
  left: 0;
  z-index: 99;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;
  height: 30px;
  background: #222528;
  box-shadow: 0 -1px #000;
  transition: 0.3s ease;
  user-select: none;
  .toggle-thumbnail {
    padding-right: 8px;
    color: #bcc9d4;
    cursor: pointer;
  }
}

.toolbox-hide {
  height: 0;
  overflow: hidden;
}
