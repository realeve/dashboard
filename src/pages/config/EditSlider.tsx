import React, { useEffect, useState } from 'react';
import styles from './EditSlider.less';
import classnames from 'classnames';

import ToolMenu from './ToolMenu';

export const rangeCfg = { min: 0.3, max: 1.5, step: 0.1 };
export const InputRange = props => {
  const percent = props.value / props.max;
  return (
    <input
      type="range"
      className={styles['input-range']}
      {...props}
      style={{
        background: `linear-gradient(to right, rgb(0, 251, 255), rgb(0, 176, 255) ${percent}%, rgb(38, 42, 53) ${percent *
          100}%, rgb(38, 42, 53))`,
      }}
    />
  );
};

export default ({ editor, onMenuChange, curTool, zoom, onZoom }) => {
  return (
    <div className={styles['edit-slider']}>
      {editor && <ToolMenu curTool={curTool} editor={editor} onSelect={onMenuChange} />}
      <div className={styles['datav-slider']}>
        <span style={{ color: '#bcc9d4', width: 35, fontSize: 12, textAlign: 'right' }}>
          {Math.floor(zoom * 100)} %
        </span>
        <i
          className="datav-icon datav-font icon-zoom-out slider-icon zoom-out "
          onClick={e => {
            let nextVal = Math.max(rangeCfg.min, zoom - 0.1);
            nextVal = Number(nextVal.toFixed(1));
            onZoom(nextVal);
          }}
        />
        <InputRange
          {...rangeCfg}
          value={zoom}
          onChange={e => {
            onZoom(Number(e.target.value));
          }}
        />
        <i
          className="datav-icon datav-font icon-zoom-in slider-icon zoom-in"
          onClick={e => {
            let nextVal = Math.min(rangeCfg.max, zoom + 0.1);
            nextVal = Number(nextVal.toFixed(1));
            onZoom(nextVal);
          }}
        />
      </div>
    </div>
  );
};
