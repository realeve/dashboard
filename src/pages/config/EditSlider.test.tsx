import React from 'react';
import EditSlider from './EditSlider';

import { create } from 'react-test-renderer';

test('编辑器右下角缩放', () => {
  let config = {
    zoom: 1,
    showConfig: true,
    hide: false,
    curTool: 'MoveTool',
    editor: null,
    onZoom: (e) => e,
    onToggleThumb: (e) => e,
    onMenuChange: (e) => e,
  };
  let root = create(<EditSlider {...config} />);
  expect(root.toJSON()).toMatchSnapshot();
  config = { ...config, zoom: 1.5, showConfig: false, hide: true };
  root.update(<EditSlider {...config} />);
  expect(root.toJSON()).toMatchSnapshot();

  config = { ...config, zoom: 1.5, showConfig: true, editor: <a>1</a>, curTool: 'hand' };
  root.update(<EditSlider {...config} />);
  expect(root.toJSON()).toMatchSnapshot();
});
