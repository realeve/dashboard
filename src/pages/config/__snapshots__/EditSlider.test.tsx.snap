// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`编辑器右下角缩放 1`] = `
<div
  className=""
  style={
    Object {
      "width": "calc(100% - 332px)",
    }
  }
>
  <div
    className=""
  >
    <span
      style={
        Object {
          "color": "#bcc9d4",
          "fontSize": 12,
          "textAlign": "right",
          "width": 50,
        }
      }
    >
      100
       %
    </span>
    <i
      className="datav-icon datav-font icon-zoom-out slider-icon zoom-out "
      onClick={[Function]}
    />
    <div>
      <input
        className=""
        disabled={false}
        max={2}
        min={0.4}
        onChange={[Function]}
        readOnly={false}
        step={0.1}
        style={
          Object {
            "background": "linear-gradient(to right, rgb(0, 251, 255), rgb(0, 176, 255) 0.5%, rgb(38, 42, 53) 50%, rgb(38, 42, 53))",
          }
        }
        type="range"
        value={1}
      />
    </div>
    <i
      className="datav-icon datav-font icon-zoom-in slider-icon zoom-in"
      onClick={[Function]}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <i
      className="datav-icon datav-font icon-viewport"
    />
  </div>
</div>
`;

exports[`编辑器右下角缩放 2`] = `
<div
  className="undefined"
  style={Object {}}
>
  <div
    className=""
  >
    <span
      style={
        Object {
          "color": "#bcc9d4",
          "fontSize": 12,
          "textAlign": "right",
          "width": 50,
        }
      }
    >
      100
       %
    </span>
    <i
      className="datav-icon datav-font icon-zoom-out slider-icon zoom-out "
      onClick={[Function]}
    />
    <div>
      <input
        className=""
        disabled={false}
        max={2}
        min={0.4}
        onChange={[Function]}
        readOnly={false}
        step={0.1}
        style={
          Object {
            "background": "linear-gradient(to right, rgb(0, 251, 255), rgb(0, 176, 255) 0.5%, rgb(38, 42, 53) 50%, rgb(38, 42, 53))",
          }
        }
        type="range"
        value={1}
      />
    </div>
    <i
      className="datav-icon datav-font icon-zoom-in slider-icon zoom-in"
      onClick={[Function]}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <i
      className="datav-icon datav-font icon-viewport"
    />
  </div>
</div>
`;

exports[`编辑器右下角缩放 3`] = `
<div
  className="undefined"
  style={
    Object {
      "width": "calc(100% - 332px)",
    }
  }
>
  <div
    className="scena-menu"
  >
    <div
      className="scena-icon scena-selected"
      onClick={[Function]}
      title="选择工具：选择屏幕中的组件"
    >
      <svg
        viewBox="0 0 80 80"
      >
        <path
          d="M 21,21 L 35,60 L 40,44 L 54,58 A 3,3 0,0,0, 58,54 L 44,40 L 60,35 L 21,21Z"
          fill="#222"
          stroke="#eee"
          strokeLinejoin="round"
          strokeWidth="3"
          style={
            Object {
              "transform": "rotate(10deg)",
              "transformOrigin": "42px 42px",
            }
          }
        />
      </svg>
    </div>
    <div
      className="scena-icon "
      onClick={[Function]}
      title="移动工具：移动屏幕位置"
    >
      <svg
        fill="#fff"
        focusable="false"
        height="1em"
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M909.3 506.3L781.7 405.6a7.23 7.23 0 00-11.7 5.7V476H548V254h64.8c6 0 9.4-7 5.7-11.7L517.7 114.7a7.14 7.14 0 00-11.3 0L405.6 242.3a7.23 7.23 0 005.7 11.7H476v222H254v-64.8c0-6-7-9.4-11.7-5.7L114.7 506.3a7.14 7.14 0 000 11.3l127.5 100.8c4.7 3.7 11.7.4 11.7-5.7V548h222v222h-64.8c-6 0-9.4 7-5.7 11.7l100.8 127.5c2.9 3.7 8.5 3.7 11.3 0l100.8-127.5c3.7-4.7.4-11.7-5.7-11.7H548V548h222v64.8c0 6 7 9.4 11.7 5.7l127.5-100.8a7.3 7.3 0 00.1-11.4z"
        />
      </svg>
    </div>
    <div
      className="scena-icon "
      onClick={[Function]}
      title="画布居中"
    >
      <svg
        fill="#fff"
        focusable="false"
        height="1em"
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
        />
      </svg>
    </div>
    <div
      className="scena-icon "
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 80 80"
      >
        <g
          transform="matrix(0.987601,0,0,0.987601,-1.97306,0.554734)"
        >
          <path
            d="M64.286,17.81L20.714,17.81L20.714,29.56L29.214,23L39.262,23L39.262,55.476L27.77,61.262L27.77,62.071L57.23,62.071L57.23,61.262L45.738,55.476L45.738,23L55.786,23L64.286,29.56L64.286,17.81Z"
            style={
              Object {
                "fill": "white",
              }
            }
          />
        </g>
      </svg>
    </div>
    <div
      className="scena-icon "
      onClick={[Function]}
    >
      <div
        className="scena-i"
        style={
          Object {
            "width": 28,
          }
        }
        title="快捷键(s)"
      >
        <div
          className="scena-keyboard"
        >
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-key"
          />
          <div
            className="scena-space"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <span
      style={
        Object {
          "color": "#bcc9d4",
          "fontSize": 12,
          "textAlign": "right",
          "width": 50,
        }
      }
    >
      150
       %
    </span>
    <i
      className="datav-icon datav-font icon-zoom-out slider-icon zoom-out "
      onClick={[Function]}
    />
    <div>
      <input
        className=""
        disabled={false}
        max={2}
        min={0.4}
        onChange={[Function]}
        readOnly={false}
        step={0.1}
        style={
          Object {
            "background": "linear-gradient(to right, rgb(0, 251, 255), rgb(0, 176, 255) 0.75%, rgb(38, 42, 53) 75%, rgb(38, 42, 53))",
          }
        }
        type="range"
        value={1.5}
      />
    </div>
    <i
      className="datav-icon datav-font icon-zoom-in slider-icon zoom-in"
      onClick={[Function]}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <i
      className="datav-icon datav-font icon-viewport"
    />
  </div>
</div>
`;
